src/parser/bgp/messages.rs

```diff
@@ -111,16 +103,17 @@ pub fn parse_bgp_notification_message(
 ///
 /// The parsing of BGP OPEN messages also includes decoding the BGP capabilities.
 pub fn parse_bgp_open_message(input: &mut Bytes) -> Result<BgpOpenMessage, ParserError> {
-    input.has_n_remaining(10)?;
-    let version = input.get_u8();
-    let asn = Asn::new_16bit(input.get_u16());
-    let hold_time = input.get_u16();
+    input.require_n_remaining(10, "BGP open message header")?;
+    let version = input.read_u8()?;
+    let asn = Asn::new_16bit(input.read_u16()?);
+    let hold_time = input.read_u16()?;
 
     let sender_ip = input.read_ipv4_address()?;
-    let opt_params_len = input.get_u8();
+    let opt_params_len = input.read_u8()?;
 
     // let pos_end = input.position() + opt_params_len as u64;
     if input.remaining() != opt_params_len as usize {
+        // TODO: This seems like it should become a hard error
```

Comment:
Yes. But it shouldn't stop the parsing of messages after this.

Reply:
We could add the same error that you introduced previously (`InconsistentFieldLength`) here.